### DISCUSSION TITLE
kube-rs in production

### DISCUSSION THREAD

**Discussion by jmintb** [kube-rs/kube#1064] [General]
I am strongly considering using kube-rs in an [MLOps](https://github.com/TeaInSpace/ame) tool which will probably end up running in production within the next year. As far as I can tell the project is in a good state and already has a few non trivial adopters. 

I have two questions:
1. Would you recommend against using kube-rs in production?
2. Is there anything I should be aware of before using kube-rs in a production setting?

**Comment by lfrancke** [comment]
I'm answering as a heavy user of the library (and minor contributor in the past).

1. No
2. Things change every once in a while in breaking ways

To elaborate:
1. The team behind kube-rs has done/is doing a tremendous job @clux and the other maintainers are awesome on jumping on issues, releases happen frequently, new features etc. are merged all the times and the project is growing in maturity along all axes (feature completeness, documentation, code maturity etc.). We've been using it in production for almost two years now.
2. There are breaking changes but the changelogs are usually elaborate, there are good reasons for it and while I don't have data to back it up I have a feeling the breaking changes are slowing down.

**Reply by jmintb** [reply]
Awesome thanks, that sounds good!
